### PR TITLE
quaternion sensor apply refactor

### PR DESCRIPTION
**Pull request by orneryd**
This refactor saves ~6% CPU on IMUF targets and ~2% on non-IMUF targets. 

needs tested and verified

### CODE REVIEW

File: src/main/flight/imu.c [hunk 1]
```diff
@@ -201,86 +201,101 @@ static float imuUseFastGains(void) {
...
+    return -(float)sin_approx(courseOverGround) * (1.0f - 2.0f * qpAttitude.yy - 2.0f * qpAttitude.zz) - cos_approx(courseOverGround) * (2.0f * (qpAttitude.xy - -qpAttitude.wz));   
+}
 
-        // William Premerlani and Paul Bizard, Direction Cosine Matrix IMU - Eqn. 22-23
-        // (Rxx; Ryx) - measured (estimated) heading vector (EF)
-        // (cos(COG), sin(COG)) - reference heading vector (EF)
-        // error is cross product between reference heading and estimated heading (calculated in EF)
-        const float ez_ef = - sin_approx(courseOverGround) * (1.0f - 2.0f * qpAttitude.yy - 2.0f * qpAttitude.zz) - cos_approx(courseOverGround) * (2.0f * (qpAttitude.xy - -qpAttitude.wz));
+FAST_CODE void applySensorCorrection(quaternion *vError){
+    UNUSED(vError);
```
Comment (kidBrazil) [inline src/main/flight/imu.c]: Implicit declaration is not allowed and is causing Trav to throw error on the TEST build

File: src/main/flight/imu.c [hunk 2]
```diff
@@ -344,46 +359,11 @@ STATIC_UNIT_TESTED void imuUpdateEulerAngles(void) {
...
-        }
-    }
-#endif
-
 #if defined(SIMULATOR_BUILD) && defined(SKIP_IMU_CALC)
     UNUSED(imuMahonyAHRSupdate);
-    UNUSED(useAcc);
-    UNUSED(useMag);
-    UNUSED(useCOG);
     UNUSED(courseOverGround);
```
Comment (kidBrazil) [inline src/main/flight/imu.c]: courseOverGround is also throwing errors on some targets. Probably some ifdef missing to isolate the feature.

File: src/main/flight/imu.c [hunk 3]
```diff
@@ -201,86 +201,101 @@ static float imuUseFastGains(void) {
...
+}
 
-    if (useCOG) {
-        // Use raw heading error (from GPS or whatever else)
-        while (courseOverGround >  M_PIf) courseOverGround -= (2.0f * M_PIf);
-        while (courseOverGround < -M_PIf) courseOverGround += (2.0f * M_PIf);
+FAST_CODE float calcMagError(quaternion *vMag){
+    quaternionNormalize(vMag);
+    
+    // (hx; hy; 0) - measured mag field vector in EF (assuming Z-component is zero)
```
Comment (digitalentity) [inline src/main/flight/imu.c]: If you are taking my code from INAV giving some credit would be nice, don't you think?

Comment (adrianmiriuta) [reply src/main/flight/imu.c]: @digitalentity 
Karma is a bitch  (GPL)

File: src/main/flight/imu.c (hunk 2 above)
Comment (kidBrazil) [inline src/main/flight/imu.c]: image

Is causing issues with some of the targets on the final group. Probably ifdef again.

**Pull request merged**